types/src/traits.rs

```diff
@@ -0,0 +1,40 @@
+use crate::client::Subscription;
+use crate::error::Error;
+use crate::jsonrpc::{DeserializeOwned, Params};
+use alloc::string::String;
+use async_trait::async_trait;
+
+/// Basic `JSONRPC` client that can make requests, notifications and subscriptions.
+#[async_trait]
+pub trait Client {
```

Comment:
one annoyance is that this trait can't be used as a trait object because of the generics (e.g, tried porting this to subxt)

we could change everything to concrete types and return `JsonValue`  instead of `T: DeserializeOwned `but it's really inelegant IMHO.